### ISSUE TITLE
AtomicOrbitals Fails for Fractional Compositions

### ISSUE THREAD

**Issue by WardLT** [hackingmaterials/matminer#301]
AtomicOrbitals gives an `IndexError` when you provide it a fractional composition (example below). Is this a bug, or me not understanding how AtomicOrbitals works? 

If it is not the result of some unit cell dependence (e.g., "NaCl" is different than "Na0.5Cl0.5"), we should make this class return the same value for any compositions with the same element fractions.

```
>>> AtomicOrbitals().featurize(Composition('NaCl'))
['p', 'Cl', -0.32038, 's', 'Na', -0.103415, 0.216965]
>>> AtomicOrbitals().featurize(Composition('Na0.5Cl0.5'))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "c:\users\ward6\documents\code\python\matminer\matminer\featurizers\composition.py", line 322, in featurize
    homo_lumo = MolecularOrbitals(string_comp).band_edges
  File "c:\users\ward6\documents\code\python\pymatgen\pymatgen\core\molecular_orbitals.py", line 60, in __init__
    self.band_edges = self.obtain_band_edges()
  File "c:\users\ward6\documents\code\python\pymatgen\pymatgen\core\molecular_orbitals.py", line 102, in obtain_band_edges
    homo = partial_filled[-1]
IndexError: list index out of range
```

**Comment by computron** [comment]
The method should give you the same result whether you put in NaCl or Na0.5Cl0.5

@dyllamt can you comment?

Also, when fixed, a quick unit test should be added to make sure this feature doesn't break in the future

**Comment by dyllamt** [comment]
It looks like fractional compositions were not supported for this featurizer at all, because the underlying Pymatgen class `MolecularOrbitals` assumes that your composition contains integer numbers of atoms (i.e. Na2Cl2 would give the same result as NaCl, but Na0.5Cl0.5 is not supported at all).

I added `get_integer_formula_and_factor` to the beginning of the `featurize` function to sanitize the input of the featurizer. This ensures that the composition passed to `MolecularOrbitals` has integer subscripts.

I also added a test to ensure that two different fractional compositions (NaCl v.s. Na0.5Cl0.5) return the same features. 

Pull request #303
